syscontainers: refactor duplicated info reading

peterbaouoft opened the pull request:
Removal of duplicated code sections for reading info fields from multiple functions.

- This is one of my first attempt of doing code refactoring, if there is anything I did horribly wrong, please tell me asap =)

- This refactoring is from part of https://github.com/projectatomic/atomic/issues/1149, it refactored duplicated code sections for installed files fields

- Shouldn't have any functionality change here, will see how the test goes (test seems passing locally)

peterbaouoft commented:
From storage test log, it should be a flake? (me guessing) 
```
+ atomic --debug storage modify --add-device /dev/vdb
Return Code: 1
ERROR: Failed to unmount /var/lib/docker /var/lib/docker/containers /var/lib/docker/overlay2

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/Atomic/storage.py", line 205, in modify
    util.check_output(["docker-storage-setup"], stderr=subprocess.STDOUT)
  File "/usr/lib/python3.6/site-packages/Atomic/util.py", line 231, in check_output
    return subprocess.check_output(cmd, env=env, stdin=stdin, stderr=stderr, close_fds=True)
  File "/usr/lib64/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib64/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker-storage-setup']' returned non-zero exit status 1.
```

giuseppe commented:
bot, retest this please

giuseppe commented on Atomic/syscontainers.py:
```diff
@@ -1114,7 +1133,9 @@ def update_container(self, name, setvalues=None, rebase=None):
             raise ValueError("Cannot find a configured OSTree repo")
 
-        path = os.path.join(self._get_system_checkout_path(), name)
+        checkout = self._get_system_checkout_path()
```
could you rename it to `system_checkout_path`?

peterbaouoft replied:
yup, sure, thanks for the review =)

giuseppe commented on Atomic/syscontainers.py:
```diff
@@ -1622,13 +1630,14 @@ def uninstall(self, name):
             return
 
-        if not os.path.exists(os.path.join(self._get_system_checkout_path(), name)):
+        checkout = self._get_system_checkout_path()
```
same here

peterbaouoft commented:
Addressed the comments above. Also, the call method was no longer via `self._gather_installed_files_info`. 

Instead, it is `SystemContainers._gather_installed_files_info` (static method calls). I think it might be better to have this factored out function to be static method and called using `CLASS.METHODNAME`, but interested in feedback =).

giuseppe commented on Atomic/syscontainers.py:
```diff
@@ -1179,23 +1193,17 @@ def rollback(self, name):
             raise ValueError("Error: Cannot find a previous deployment to rollback located at %s" % destination)
 
-        installed_files = None
+        info, rpm_installed, installed_files_checksum = SystemContainers._gather_installed_files_info(system_checkout_path, name)
```
we don't really follow this line for long lines.  I think it is fine to leave the full line.

After this, I think the patch is good to go!

peterbaouoft replied:
? sorry, can you explain a bit more? (me guessing) Are you suggesting the following:
```
tuple = SystemContainers._gather_installed_files_info(system_checkout_path, name)
info = tuple[0]
rpm_installed = tuple[1]
installed_files_checksum = tuple[2]
```
?

giuseppe replied:
no nevermind it was just a mistake on my side.

giuseppe commented:
@rh-atomic-bot r+ 5b2e324
